Sources/PostgresNIO/Connection/PostgresConnection+Configuration.swift

```diff
@@ -0,0 +1,318 @@
+            /// SNI is disabled.
+            ///
+            /// - Warning: This name is not validated in any way; in particular, there is no attempt to check that it
+            ///   matches the server's TLS certificate, nor is it in any way impacted by the validation setting in the
+            ///   TLS configuration (see ``PostgresConnection/Configuration/Server-swift.struct/TLS-swift.struct``).
+            public let tlsServerName: String?
+            
+            /// Whether the connection is required to provide backend key data (internal Postgres stuff).
+            ///
+            /// This property is provided for compatibility with Amazon RDS Proxy, which requires it to be `false`.
+            /// If you are not using Amazon RDS Proxy, you should probably leave this set to `true` (the default).
+            public let requireBackendKeyData: Bool
+            
+            /// Configure various options for a connection.
+            ///
+            /// Most users should not need to adjust the defaults.
+            ///
+            /// - Parameters:
+            ///   - connectTimeout: See ``connectTimeout``. Defaults to 10 seconds.
+            ///   - tlsServerName: See ``tlsServerName``. Default is `nil`.
+            ///   - requireBackendKeyData: See ``requireBackendKeyData``. Default is `true`.
+            public init(
+                connectTimeout: TimeAmount = .seconds(10),
+                tlsServerName: String? = nil,
+                requireBackendKeyData: Bool = true
+            ) {
+                self.connectTimeout = connectTimeout
+                self.tlsServerName = tlsServerName
+                self.requireBackendKeyData = requireBackendKeyData
+            }
+        }
+
+        /// Endpoint information for establishing a communication channel.
+        ///
+        /// See ``Server-swift.struct``.
+        public let server: Server
+        
+        /// Authentication properties to send during the startup auth handshake.
+        ///
+        /// See ``Authentication-swift.struct``.
+        public var authentication: Authentication // TODO: Make this immutable
+        
+        /// Options for handling the communication channel. Most users don't need to change these.
+        ///
+        /// See ``Options-swift.struct``.
+        public let options: Options
+        
+        /// Create a complete connection configuration.
+        ///
+        /// - Parameters:
+        ///   - server: See ``Server-swift.struct``.
+        ///   - authentication: See ``Authentication-swift.struct``.
+        ///   - options: See ``Options-swift.struct``. Most users do not need to adjust any of these.
+        public init(
+            server: Server,
+            authentication: Authentication,
+            options: Options = .init()
+        ) {
+            self.server = server
+            self.authentication = authentication
+            self.options = options
+        }
```

Comment:
In an ideal case this is just an init and users do all changes through setting properties. See TLSConfiguration for comparison.